### PR TITLE
Added necessary files to create an image for stats.

### PR DESCRIPTION
**Pull request by tebadi**

### CODE REVIEW

File: docker/libs/stats/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+ENV LC_ALL=C.UTF-8
+ENV DEBIAN_FRONTEND=noninteractive
+
+# Do the apt install process, including more recent Postgres/PostGIS
+RUN apt-get update && apt-get install -y wget gnupg \
+    && rm -rf /var/lib/apt/lists/*
+RUN wget --quiet -O - https://www.postgresql.org/media/keys/ACCC4CF8.asc | \
+    apt-key add - \
+    && echo "deb http://apt.postgresql.org/pub/repos/apt/ bionic-pgdg main" \
+    >> /etc/apt/sources.list.d/postgresql.list
```
Comment (alexgleith) [inline docker/libs/stats/Dockerfile]: Can we please test without this step as I don't think it's needed.

File: docker/libs/stats/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+FROM opendatacube/geobase:runner
+COPY --from=env_builder /env /env
+
+ENV PATH="/env/bin:${PATH}"
+ENV LC_ALL=C.UTF-8
+ENV LANG=C.UTF-8
+
+# Copy Datacube bootstrapping script
+# COPY /assets/bootstrap-odc.sh /code/bootstrap-odc.sh
```
Comment (alexgleith) [inline docker/libs/stats/Dockerfile]: Let's get rid of these, please.

File: docker/libs/stats/docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+services:
+  # Start docker container for PostgreSQL to mock RDS
+  dc-db:
+    image: odc-stats-v01
+    ports:
+      - "5434:5432"
+  maildev:
+    image: maildev/maildev
+    ports:
+      - "1080:80"
```
Comment (alexgleith) [inline docker/libs/stats/docker-compose.yml]: This looks like it came from Datacube Alchemist.

Maildev is for pretending to be a mailserver that you can send emails to.

Also, the `dc-db` should be postgres.

Please copy this: https://github.com/opendatacube/cube-in-a-box and test it before committing.

File: docker/libs/stats/requirements.txt [hunk 4]
```diff
@@ -0,0 +1,21 @@
+--pre
```
Comment (alexgleith) [inline docker/libs/stats/requirements.txt]: I think we can remove `--pre`

File: docker/libs/stats/docker-compose.yml (hunk 3 above)
Comment (alexgleith) [reply docker/libs/stats/docker-compose.yml]: Basically, we should have two services:

- postgres: this is the postgres DB
- statistician: this is the Docker image we're building (so use `build: .` and not an image)

File: docker/libs/stats/requirements.txt [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+--extra-index-url https://packages.dea.ga.gov.au/
+datacube[performance,s3]==1.8.3
+aiobotocore[boto3,awscli]==1.1.1
+aiohttp==3.6.2
+pandas==1.0.5
+tqdm==4.48.2
+digitalearthau
+thredds-crawler
```
Comment (alexgleith) [inline docker/libs/stats/requirements.txt]: Nope

File: docker/libs/stats/requirements.txt [hunk 6]
```diff
@@ -0,0 +1,21 @@
+--pre
+--extra-index-url https://packages.dea.ga.gov.au/
+datacube[performance,s3]==1.8.3
+aiobotocore[boto3,awscli]==1.1.1
+aiohttp==3.6.2
+pandas==1.0.5
+tqdm==4.48.2
+digitalearthau
```
Comment (alexgleith) [inline docker/libs/stats/requirements.txt]: Don't think we need this.

File: docker/libs/stats/requirements.txt [hunk 7]
```diff
@@ -0,0 +1,21 @@
+--pre
+--extra-index-url https://packages.dea.ga.gov.au/
+datacube[performance,s3]==1.8.3
+aiobotocore[boto3,awscli]==1.1.1
+aiohttp==3.6.2
+pandas==1.0.5
+tqdm==4.48.2
```
Comment (alexgleith) [inline docker/libs/stats/requirements.txt]: Not sure about these two?

File: docker/libs/stats/requirements.txt [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+digitalearthau
+thredds-crawler
+wget
+requests
+odc-aws
+odc-aio
+odc-stats
+odc-thredds
```
Comment (alexgleith) [inline docker/libs/stats/requirements.txt]: Nope

**Review by alexgleith — COMMENTED** [review]
This needs to be in requirements: `git+https://github.com/opendatacube/datacube-index.git`

See other comments.

File: docker/libs/stats/Dockerfile (hunk 1 above)
Comment (tebadi) [reply docker/libs/stats/Dockerfile]: It worked without these.  I've deleted these.

File: docker/libs/stats/Dockerfile (hunk 2 above)
Comment (tebadi) [reply docker/libs/stats/Dockerfile]: Done.

File: docker/libs/stats/docker-compose.yml (hunk 3 above)
Comment (tebadi) [reply docker/libs/stats/docker-compose.yml]: I've addressed these.

File: docker/libs/stats/requirements.txt (hunk 4 above)
Comment (tebadi) [reply docker/libs/stats/requirements.txt]: Deleted.

File: docker/libs/stats/requirements.txt (hunk 5 above)
Comment (tebadi) [reply docker/libs/stats/requirements.txt]: Deleted pandas, digitalearthau, thredds-crawler, digitalearthau.

File: docker/libs/stats/requirements.txt (hunk 6 above)
Comment (tebadi) [reply docker/libs/stats/requirements.txt]: Done.

File: docker/libs/stats/requirements.txt (hunk 7 above)
Comment (tebadi) [reply docker/libs/stats/requirements.txt]: tqdm is required by the indexing.

File: docker/libs/stats/requirements.txt (hunk 8 above)
Comment (tebadi) [reply docker/libs/stats/requirements.txt]: Removed odc-thredds.

**Comment by Kirill888** [comment]
I'd like to have access to `tiffinfo` command inside this docker,
It's in this apt package `libtiff-tools`.

File: docker/stats/requirements.txt [hunk 9]
```diff
@@ -6,6 +6,7 @@ wget
 requests
 odc-aws
 odc-aio
+odc-algo
```
Comment (Kirill888) [inline docker/stats/requirements.txt:9]: The issue is not missing `odc-algo` issue is missing `hdstats` which `odc-algo` is using without declaring dependency on it.

**Pull request closed without merging**